code/ship/ship.cpp

```diff
@@ -11298,8 +11299,9 @@ int ship_fire_primary(object * obj, int force, bool rollback_shot)
 		return 0;
 	}	
 
-	// If the primaries have been locked, bail
-	if (shipp->flags[Ship_Flags::Primaries_locked])
+	// If the primaries have been locked, bail. 
+	// Unless we're dealing with the player and their ship has the flag set to allow fail sounds when firing locked primaries.
+	if (shipp->flags[Ship_Flags::Primaries_locked] && ! (obj == Player_obj && shipp->flags[Ship_Flags::Fail_sound_locked_primary]))
 	{
```

Comment:
For example:
```suggestion
	// If the primaries have been locked, bail.
	if (shipp->flags[Ship_Flags::Primaries_locked]) {
		// If we're dealing with the player and their ship has the flag set to allow fail sounds when firing locked primaries, do that first.
		if (obj == Player_obj && shipp->flags[Ship_Flags::Fail_sound_locked_primary]) {
			ship_maybe_do_primary_primary_fail_sound_hud(false);
		}
```

Reply:
That's what I tried at first. When I added the code in that spot, the sound played every 50ms regardless of the weapon's fire-rate.  
Adding it afterwards ensures the fire rate has already been handled.

Reply:
In that case, there's still an earlier location you can put it: around line 11527 (on master, anyway) there's a block that starts like this:
```cpp
		// Here is where we check if weapons subsystem is capable of firing the weapon.
		// Note that we can have partial bank firing, if the weapons subsystem is partially
		// functional, which should be cool.  		
		if ( ship_weapon_maybe_fail(shipp) && !force) {
```
This does something similar (play the gun-jammed sound effect and stop the firing action if the weapons subsystem is damaged), and avoids wasting its time calculating auto-aim and whatnot. Additionally, if you leave the code where it is, it only applies to non-beam primaries, while the block mentioned above happens before that conditional split.